packages/address-book-controller/package.json

```diff
@@ -42,6 +46,9 @@
     "typedoc-plugin-missing-exports": "^0.22.6",
     "typescript": "~4.6.3"
   },
+  "peerDependencies": {
+    "@metamask/preferences-controller": "workspace:^"
```

Comment:
`preferences-controller` is being added as a peer dependency because we only use types from this package and not runtime code.

Reply:
Type dependencies should be in `dependencies` if they're exported or referenced in exported types, or `devDependencies` otherwise.

`peerDependencies` is for the situation where the project using this package needs to separately install something for this to work correctly, sometimes with a specific configuration/setup. Like with the controller messenger, for example. Or when a package is a plugin for a system that you need to ensure is also installed.

Reply:
Ah, I keep getting confused about what ends up in the type definition files. I was thinking that the type from `preferences-controller` would go away somehow but it does stay. Which makes sense now that I think about it. 

Really, I was trying to avoid having `address-book-controller` depend on `preferences-controller`, because the whole point of having AddressBookController take `onPreferencesStateChange` is that it doesn't have to actually care about PreferencesController. Now it still does. Hmm...

Reply:
Perhaps we don't need the preferences controller state type? It looks like we're only using the frequent RPC list. We could inline just the type definition for that property.

Reply:
On a similar note, maybe we could consider making the contstructor parameter something like `onRpcListChange` so that we wouldn't need to change it when migrating the state. Then it would make even more sense not to include the preferences controller state type.

Reply:
Okay. Good suggestions. I'll try `onFrequentRpcListChange`.